File: fabric/src/main/java/wile/redstonepen/gametest/TrackGameTests.java
package wile.redstonepen.gametest;

import net.minecraft.gametest.framework.GameTest;
import net.minecraft.gametest.framework.GameTestHelper;
import wile.redstonepen.gametestcommon.TrackTests;

public class TrackGameTests
{
  private static final String TEMPLATE = "redstonepen:relay_activates_from_redstone";

  public TrackGameTests() {}

  @GameTest(template = TEMPLATE, timeoutTicks = 20)
  public static void trackStoresSeededPowerRoute(GameTestHelper helper)
  { TrackTests.trackStoresSeededPowerRoute(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 20)
  public static void trackReplacesSeededRouteWhenPowerClears(GameTestHelper helper)
  { TrackTests.trackReplacesSeededRouteWhenPowerClears(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 10)
  public static void writenbtSyncOmitsNetsList(GameTestHelper helper)
  { TrackTests.writenbtSyncOmitsNetsList(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 10)
  public static void writenbtFullIncludesNetsList(GameTestHelper helper)
  { TrackTests.writenbtFullIncludesNetsList(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 10)
  public static void readnbtRestoresStateFlagsAndNets(GameTestHelper helper)
  { TrackTests.readnbtRestoresStateFlagsAndNets(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 10)
  public static void readnbtAcceptsCorruptNetsListWithoutThrowing(GameTestHelper helper)
  { TrackTests.readnbtAcceptsCorruptNetsListWithoutThrowing(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void addWireFlagsRecordsOnlyNewBits(GameTestHelper helper)
  { TrackTests.addWireFlagsRecordsOnlyNewBits(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void getWireFlagsReadsIndividualBits(GameTestHelper helper)
  { TrackTests.getWireFlagsReadsIndividualBits(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void setSidePowerAndGetSidePowerRoundTrip(GameTestHelper helper)
  { TrackTests.setSidePowerAndGetSidePowerRoundTrip(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void hasVanillaRedstoneConnectionReadsBitsAndConnectorMask(GameTestHelper helper)
  { TrackTests.hasVanillaRedstoneConnectionReadsBitsAndConnectorMask(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void updateAllPowerValuesOnIsolatedTrackReturnsMap(GameTestHelper helper)
  { TrackTests.updateAllPowerValuesOnIsolatedTrackReturnsMap(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void updateAllPowerValuesOnTrackWithSeededNetReturnsMap(GameTestHelper helper)
  { TrackTests.updateAllPowerValuesOnTrackWithSeededNetReturnsMap(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void handleShapeUpdateForAirNeighborReturns(GameTestHelper helper)
  { TrackTests.handleShapeUpdateForAirNeighborReturns(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void handleShapeUpdateMovingFlagSkipsRecursion(GameTestHelper helper)
  { TrackTests.handleShapeUpdateMovingFlagSkipsRecursion(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void handleShapeUpdateRedstoneBlockNeighborSkipsConnectionRefresh(GameTestHelper helper)
  { TrackTests.handleShapeUpdateRedstoneBlockNeighborSkipsConnectionRefresh(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void getShapeReflectsWireFlags(GameTestHelper helper)
  { TrackTests.getShapeReflectsWireFlags(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void emptyTrackShapeIsEmpty(GameTestHelper helper)
  { TrackTests.emptyTrackShapeIsEmpty(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void getCollisionShapeIsAlwaysEmpty(GameTestHelper helper)
  { TrackTests.getCollisionShapeIsAlwaysEmpty(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void getSignalForUnpoweredTrackIsZero(GameTestHelper helper)
  { TrackTests.getSignalForUnpoweredTrackIsZero(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void getDirectSignalForUnpoweredTrackIsZero(GameTestHelper helper)
  { TrackTests.getDirectSignalForUnpoweredTrackIsZero(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void canSurviveAlwaysTrue(GameTestHelper helper)
  { TrackTests.canSurviveAlwaysTrue(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void shouldCheckWeakPowerIsFalse(GameTestHelper helper)
  { TrackTests.shouldCheckWeakPowerIsFalse(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void canConnectRedstoneFalseForUnconnectedTrack(GameTestHelper helper)
  { TrackTests.canConnectRedstoneFalseForUnconnectedTrack(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void canConnectRedstoneFalseForNullSide(GameTestHelper helper)
  { TrackTests.canConnectRedstoneFalseForNullSide(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void propagatesSkylightDownDependsOnWaterlogged(GameTestHelper helper)
  { TrackTests.propagatesSkylightDownDependsOnWaterlogged(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void useShapeForLightOcclusionTrue(GameTestHelper helper)
  { TrackTests.useShapeForLightOcclusionTrue(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void getRenderShapeIsAnimatedEntityBlock(GameTestHelper helper)
  { TrackTests.getRenderShapeIsAnimatedEntityBlock(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void notifyAdjacentRunsWithoutThrowing(GameTestHelper helper)
  { TrackTests.notifyAdjacentRunsWithoutThrowing(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void dropListReturnsRedstoneDustMatchingWireCount(GameTestHelper helper)
  { TrackTests.dropListReturnsRedstoneDustMatchingWireCount(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void dropListEmptyForNoWires(GameTestHelper helper)
  { TrackTests.dropListEmptyForNoWires(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void isPathfindableAlwaysTrue(GameTestHelper helper)
  { TrackTests.isPathfindableAlwaysTrue(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void neighborChangedDoesNotThrow(GameTestHelper helper)
  { TrackTests.neighborChangedDoesNotThrow(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void onRemoveNotifiesAdjacentWhenReplaced(GameTestHelper helper)
  { TrackTests.onRemoveNotifiesAdjacentWhenReplaced(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 10)
  public static void modifySegmentsAddThenRemoveReturnsConsume(GameTestHelper helper)
  { TrackTests.modifySegmentsAddThenRemoveReturnsConsume(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 10)
  public static void modifySegmentsAddRemoveUntilEmptyRemovesBlock(GameTestHelper helper)
  { TrackTests.modifySegmentsAddRemoveUntilEmptyRemovesBlock(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void toggleTraceWithNullPlayerDoesNotThrow(GameTestHelper helper)
  { TrackTests.toggleTraceWithNullPlayerDoesNotThrow(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void toggleTraceWithMockPlayerDoesNotThrow(GameTestHelper helper)
  { TrackTests.toggleTraceWithMockPlayerDoesNotThrow(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void connectionFlagAccessorsDoNotThrow(GameTestHelper helper)
  { TrackTests.connectionFlagAccessorsDoNotThrow(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void trackNetToStringProducesNonEmptyString(GameTestHelper helper)
  { TrackTests.trackNetToStringProducesNonEmptyString(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void getRedstonePowerZeroOnIsolatedTrack(GameTestHelper helper)
  { TrackTests.getRedstonePowerZeroOnIsolatedTrack(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void getRedstoneDustCountZeroForFreshTrack(GameTestHelper helper)
  { TrackTests.getRedstoneDustCountZeroForFreshTrack(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void getRedstoneDustCountMatchesWireFlags(GameTestHelper helper)
  { TrackTests.getRedstoneDustCountMatchesWireFlags(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void trackHasDynamicDropListTrue(GameTestHelper helper)
  { TrackTests.trackHasDynamicDropListTrue(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void trackAsItemReturnsRedstone(GameTestHelper helper)
  { TrackTests.trackAsItemReturnsRedstone(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void trackIsSignalSourceTrue(GameTestHelper helper)
  { TrackTests.trackIsSignalSourceTrue(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void trackUseWithoutItemWithPlayerDoesNotThrow(GameTestHelper helper)
  { TrackTests.trackUseWithoutItemWithPlayerDoesNotThrow(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void trackUseItemOnWithDebugStickTogglesTrace(GameTestHelper helper)
  { TrackTests.trackUseItemOnWithDebugStickTogglesTrace(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 10)
  public static void trackCanBePlacedOnFaceOfPiston(GameTestHelper helper)
  { TrackTests.trackCanBePlacedOnFaceOfPiston(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void trackCanBePlacedOnFaceOfHopper(GameTestHelper helper)
  { TrackTests.trackCanBePlacedOnFaceOfHopper(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void trackNeighborChangedWithRedstoneBlockTriggersUpdate(GameTestHelper helper)
  { TrackTests.trackNeighborChangedWithRedstoneBlockTriggersUpdate(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void trackGetRedstoneDustCountZeroWhenNoWiresLargeConfig(GameTestHelper helper)
  { TrackTests.trackGetRedstoneDustCountZeroWhenNoWiresLargeConfig(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 5)
  public static void trackReadnbtWithSflagsField(GameTestHelper helper)
  { TrackTests.trackReadnbtWithSflagsField(helper); }

  @GameTest(template = TEMPLATE, timeoutTicks = 10)
  public static void trackGetNonWireSignalFromRedstoneBlock(GameTestHelper helper)
  { TrackTests.trackGetNonWireSignalFromRedstoneBlock(helper); }
}

File: neoforge/src/main/kotlin/wile/redstonepen/gametest/TrackGameTests.kt
package wile.redstonepen.gametest

import net.minecraft.gametest.framework.GameTest
import net.minecraft.gametest.framework.GameTestHelper
import net.neoforged.neoforge.gametest.GameTestHolder
import net.neoforged.neoforge.gametest.PrefixGameTestTemplate
import wile.redstonepen.ModConstants
import wile.redstonepen.gametestcommon.TrackTests

@GameTestHolder(ModConstants.MODID)
@PrefixGameTestTemplate(false)
object TrackGameTests {
    private const val TEMPLATE = "relay_activates_from_redstone"

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 20)
    fun trackStoresSeededPowerRoute(helper: GameTestHelper) =
        TrackTests.trackStoresSeededPowerRoute(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 20)
    fun trackReplacesSeededRouteWhenPowerClears(helper: GameTestHelper) =
        TrackTests.trackReplacesSeededRouteWhenPowerClears(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 10)
    fun writenbtSyncOmitsNetsList(helper: GameTestHelper) =
        TrackTests.writenbtSyncOmitsNetsList(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 10)
    fun writenbtFullIncludesNetsList(helper: GameTestHelper) =
        TrackTests.writenbtFullIncludesNetsList(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 10)
    fun readnbtRestoresStateFlagsAndNets(helper: GameTestHelper) =
        TrackTests.readnbtRestoresStateFlagsAndNets(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 10)
    fun readnbtAcceptsCorruptNetsListWithoutThrowing(helper: GameTestHelper) =
        TrackTests.readnbtAcceptsCorruptNetsListWithoutThrowing(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun addWireFlagsRecordsOnlyNewBits(helper: GameTestHelper) =
        TrackTests.addWireFlagsRecordsOnlyNewBits(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun getWireFlagsReadsIndividualBits(helper: GameTestHelper) =
        TrackTests.getWireFlagsReadsIndividualBits(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun setSidePowerAndGetSidePowerRoundTrip(helper: GameTestHelper) =
        TrackTests.setSidePowerAndGetSidePowerRoundTrip(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun hasVanillaRedstoneConnectionReadsBitsAndConnectorMask(helper: GameTestHelper) =
        TrackTests.hasVanillaRedstoneConnectionReadsBitsAndConnectorMask(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun updateAllPowerValuesOnIsolatedTrackReturnsMap(helper: GameTestHelper) =
        TrackTests.updateAllPowerValuesOnIsolatedTrackReturnsMap(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun updateAllPowerValuesOnTrackWithSeededNetReturnsMap(helper: GameTestHelper) =
        TrackTests.updateAllPowerValuesOnTrackWithSeededNetReturnsMap(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun handleShapeUpdateForAirNeighborReturns(helper: GameTestHelper) =
        TrackTests.handleShapeUpdateForAirNeighborReturns(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun handleShapeUpdateMovingFlagSkipsRecursion(helper: GameTestHelper) =
        TrackTests.handleShapeUpdateMovingFlagSkipsRecursion(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun handleShapeUpdateRedstoneBlockNeighborSkipsConnectionRefresh(helper: GameTestHelper) =
        TrackTests.handleShapeUpdateRedstoneBlockNeighborSkipsConnectionRefresh(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun getShapeReflectsWireFlags(helper: GameTestHelper) =
        TrackTests.getShapeReflectsWireFlags(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun emptyTrackShapeIsEmpty(helper: GameTestHelper) = TrackTests.emptyTrackShapeIsEmpty(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun getCollisionShapeIsAlwaysEmpty(helper: GameTestHelper) =
        TrackTests.getCollisionShapeIsAlwaysEmpty(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun getSignalForUnpoweredTrackIsZero(helper: GameTestHelper) =
        TrackTests.getSignalForUnpoweredTrackIsZero(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun getDirectSignalForUnpoweredTrackIsZero(helper: GameTestHelper) =
        TrackTests.getDirectSignalForUnpoweredTrackIsZero(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun canSurviveAlwaysTrue(helper: GameTestHelper) = TrackTests.canSurviveAlwaysTrue(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun shouldCheckWeakPowerIsFalse(helper: GameTestHelper) =
        TrackTests.shouldCheckWeakPowerIsFalse(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun canConnectRedstoneFalseForUnconnectedTrack(helper: GameTestHelper) =
        TrackTests.canConnectRedstoneFalseForUnconnectedTrack(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun canConnectRedstoneFalseForNullSide(helper: GameTestHelper) =
        TrackTests.canConnectRedstoneFalseForNullSide(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun propagatesSkylightDownDependsOnWaterlogged(helper: GameTestHelper) =
        TrackTests.propagatesSkylightDownDependsOnWaterlogged(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun useShapeForLightOcclusionTrue(helper: GameTestHelper) =
        TrackTests.useShapeForLightOcclusionTrue(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun getRenderShapeIsAnimatedEntityBlock(helper: GameTestHelper) =
        TrackTests.getRenderShapeIsAnimatedEntityBlock(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun notifyAdjacentRunsWithoutThrowing(helper: GameTestHelper) =
        TrackTests.notifyAdjacentRunsWithoutThrowing(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun dropListReturnsRedstoneDustMatchingWireCount(helper: GameTestHelper) =
        TrackTests.dropListReturnsRedstoneDustMatchingWireCount(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun dropListEmptyForNoWires(helper: GameTestHelper) = TrackTests.dropListEmptyForNoWires(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun isPathfindableAlwaysTrue(helper: GameTestHelper) =
        TrackTests.isPathfindableAlwaysTrue(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun neighborChangedDoesNotThrow(helper: GameTestHelper) =
        TrackTests.neighborChangedDoesNotThrow(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun onRemoveNotifiesAdjacentWhenReplaced(helper: GameTestHelper) =
        TrackTests.onRemoveNotifiesAdjacentWhenReplaced(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 10)
    fun modifySegmentsAddThenRemoveReturnsConsume(helper: GameTestHelper) =
        TrackTests.modifySegmentsAddThenRemoveReturnsConsume(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 10)
    fun modifySegmentsAddRemoveUntilEmptyRemovesBlock(helper: GameTestHelper) =
        TrackTests.modifySegmentsAddRemoveUntilEmptyRemovesBlock(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun toggleTraceWithNullPlayerDoesNotThrow(helper: GameTestHelper) =
        TrackTests.toggleTraceWithNullPlayerDoesNotThrow(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun toggleTraceWithMockPlayerDoesNotThrow(helper: GameTestHelper) =
        TrackTests.toggleTraceWithMockPlayerDoesNotThrow(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun connectionFlagAccessorsDoNotThrow(helper: GameTestHelper) =
        TrackTests.connectionFlagAccessorsDoNotThrow(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun trackNetToStringProducesNonEmptyString(helper: GameTestHelper) =
        TrackTests.trackNetToStringProducesNonEmptyString(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun getRedstonePowerZeroOnIsolatedTrack(helper: GameTestHelper) =
        TrackTests.getRedstonePowerZeroOnIsolatedTrack(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun getRedstoneDustCountZeroForFreshTrack(helper: GameTestHelper) =
        TrackTests.getRedstoneDustCountZeroForFreshTrack(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun getRedstoneDustCountMatchesWireFlags(helper: GameTestHelper) =
        TrackTests.getRedstoneDustCountMatchesWireFlags(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun trackHasDynamicDropListTrue(helper: GameTestHelper) =
        TrackTests.trackHasDynamicDropListTrue(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun trackAsItemReturnsRedstone(helper: GameTestHelper) =
        TrackTests.trackAsItemReturnsRedstone(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun trackIsSignalSourceTrue(helper: GameTestHelper) = TrackTests.trackIsSignalSourceTrue(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun trackUseWithoutItemWithPlayerDoesNotThrow(helper: GameTestHelper) =
        TrackTests.trackUseWithoutItemWithPlayerDoesNotThrow(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun trackUseItemOnWithDebugStickTogglesTrace(helper: GameTestHelper) =
        TrackTests.trackUseItemOnWithDebugStickTogglesTrace(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 10)
    fun trackCanBePlacedOnFaceOfPiston(helper: GameTestHelper) =
        TrackTests.trackCanBePlacedOnFaceOfPiston(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun trackCanBePlacedOnFaceOfHopper(helper: GameTestHelper) =
        TrackTests.trackCanBePlacedOnFaceOfHopper(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun trackNeighborChangedWithRedstoneBlockTriggersUpdate(helper: GameTestHelper) =
        TrackTests.trackNeighborChangedWithRedstoneBlockTriggersUpdate(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun trackGetRedstoneDustCountZeroWhenNoWiresLargeConfig(helper: GameTestHelper) =
        TrackTests.trackGetRedstoneDustCountZeroWhenNoWiresLargeConfig(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 5)
    fun trackReadnbtWithSflagsField(helper: GameTestHelper) =
        TrackTests.trackReadnbtWithSflagsField(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 10)
    fun trackGetNonWireSignalFromRedstoneBlock(helper: GameTestHelper) =
        TrackTests.trackGetNonWireSignalFromRedstoneBlock(helper)

    @JvmStatic
    @GameTest(template = TEMPLATE, timeoutTicks = 10)
    fun verticalFaceTrackPowerPreservedAfterBottomSegmentRemoveAndReplace(helper: GameTestHelper) =
        TrackTests.verticalFaceTrackPowerPreservedAfterBottomSegmentRemoveAndReplace(helper)
}
